### PR TITLE
Added metrics for the encoding manager.

### PR DESCRIPTION
**Pull request by cody-littley**
## Why are these changes needed?

Add metrics for the encoding manager.

## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ian-shim — APPROVED** [review]
lgtm!

File: disperser/cmd/controller/main.go [hunk 1]
```diff
@@ -154,5 +176,12 @@ func RunController(ctx *cli.Context) error {
 		return fmt.Errorf("failed to start dispatcher: %v", err)
 	}
 
+	go func() {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply disperser/cmd/controller/main.go:179]: I thought about this before you commented. The core issue is that `RunController()` starts both the encoding manager and the dispatcher, and we only want to use a single metrics server that they both share.

File: disperser/controller/encoding_manager_metrics.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		batchSize:               batchSize,
+		batchDataSize:           batchDataSize,
+		batchRetryCount:         batchRetryCount,
+		batchSleepTime:          batchSleepTime,
+		failedSubmissionCount:   failSubmissionCount,
+	}
+}
+
+func (m *encodingManagerMetrics) reportBatchSubmissionLatency(duration time.Duration) {
+	m.batchSubmissionLatency.WithLabelValues().Observe(float64(duration.Nanoseconds()) / float64(time.Millisecond))
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply disperser/controller/encoding_manager_metrics.go:137]: My dislike for `Milliseconds()` is that it reports an integer value, but we are measuring things where sub-millisecond accuracy may end up being meaningful. Converting from nanoseconds is more cumbersome, but utilizes the full accuracy of the clock.

File: disperser/controller/encoding_manager_metrics.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+type encodingManagerMetrics struct {
+	batchSubmissionLatency  *prometheus.SummaryVec
+	blobHandleLatency       *prometheus.SummaryVec
+	encodingLatency         *prometheus.SummaryVec
+	putBlobCertLatency      *prometheus.SummaryVec
+	updateBlobStatusLatency *prometheus.SummaryVec
+	batchSize               *prometheus.GaugeVec
+	batchDataSize           *prometheus.GaugeVec
+	batchRetryCount         *prometheus.GaugeVec
+	batchSleepTime          *prometheus.GaugeVec
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply disperser/controller/encoding_manager_metrics.go:21]: removed

File: disperser/controller/encoding_manager_metrics.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ian-shim) [reply disperser/controller/encoding_manager_metrics.go:126]: > but we are measuring things where sub-millisecond accuracy may end up being meaningful

Not sure if this is true, but I'm fine with this

**Pull request merged**